nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -29,30 +28,25 @@ class ServiceClient
 public:
   explicit ServiceClient(
     const std::string & service_name,
-    const rclcpp::Node::SharedPtr & provided_node = rclcpp::Node::SharedPtr())
-  : service_name_(service_name)
+    const rclcpp::Node::SharedPtr & provided_node)
+  : service_name_(service_name), node_(provided_node)
   {
-    if (provided_node) {
-      node_ = provided_node;
-    } else {
-      node_ = generate_internal_node(service_name + "_Node");
-    }
-    client_ = node_->create_client<ServiceT>(service_name);
-  }
-
-  ServiceClient(const std::string & service_name, const std::string & parent_name)
-  : service_name_(service_name)
-  {
-    node_ = generate_internal_node(parent_name + std::string("_") + service_name + "_client");
-    client_ = node_->create_client<ServiceT>(service_name);
+    callback_group_ = node_->create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive,
+      false);
+    callback_group_executor_.add_callback_group(callback_group_, node_->get_node_base_interface());
+    client_ = node_->create_client<ServiceT>(
+      service_name,
+      rmw_qos_profile_services_default,
+      callback_group_);
   }
 
   using RequestType = typename ServiceT::Request;
   using ResponseType = typename ServiceT::Response;
 
   typename ResponseType::SharedPtr invoke(
     typename RequestType::SharedPtr & request,
-    const std::chrono::nanoseconds timeout = std::chrono::nanoseconds::max())
+    const std::chrono::seconds timeout = std::chrono::seconds(1))
```

Comment:
Don't change the defaults, please change back to max as default argument. 1 second is pretty arbitrary for a random system and no-timeout is a pretty reasonable default.

Reply:
> Don't change the defaults, please change back to max as default argument. 1 second is pretty arbitrary for a random system and no-timeout is a pretty reasonable default. If you `invoke` and you don't want an infinite timeout, you should include one

I agree with you, but the problem is that using the default value the service client doesn't work. Using default timeout makes it fail "service client: async_send_request failed".

Reply:
1 second is arbitrary, we don't accept arbitrary magic numbers in ROS2. If there's a default at all, it must be reasonable or provide no default at all

Reply:
Ok, using -1 as default value like `spin_until_future_complete` does works fine.